File: java_sim/simulation_188623/src/Trawa.java
import javax.swing.ImageIcon;

public class Trawa extends Roslina {

    Trawa(Swiat swiatOrganizmu, int x_pos, int y_pos, int wiek, boolean czyCzeka) {
        super(swiatOrganizmu, 0, x_pos, y_pos, wiek, czyCzeka); 
        this.ikonka = new ImageIcon("zdjecia/trawa.png");       
        this.hexIkonka = new ImageIcon("zdjecia/trawaHex.png");
    }

    String getNazwa() {
        return ("Trawa");
    }
    
    @Override
    char getZnaczek() {
        return '8'; 
    }

    @Override
    Organizm klonowanie(int posX, int posY) {
        return new Trawa(swiatOrganizmu, posX, posY, 1, true);
    }
}

File: cpp_sim/188623/Trawa.cpp
#include "Trawa.h"

Trawa::Trawa(Swiat* swiatOrganizmu, int posX, int posY, int age, bool czyCzeka)
: Roslina(swiatOrganizmu, 0, posX, posY, age, czyCzeka){
};

void Trawa::rysowanie(){
    printf("#");
};

string Trawa::getNazwa(){
    return ("Trawa");
};

Organizm *Trawa::klonowanie(int posX, int posY){
    return new Trawa(swiatOrganizmu, posX, posY, 1, true);
};

Trawa::~Trawa(){
    
};